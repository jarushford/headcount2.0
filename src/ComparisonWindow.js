import React from 'react'
import PropTypes from 'prop-types'

export default function ComparisonWindow({ compared1, compared2, comparedAvg }) {
  return (
      <div className='comparison-window'>
        <h2>{compared1.location} -- {compared1.districtAvg}</h2>
        <p>has</p>
        <h3>{comparedAvg.compared}</h3>
        <p>times the average attendance of</p>
        <h2>{compared2.location} -- {compared2.districtAvg}</h2>
      </div>
    )
}

ComparisonWindow.defaultProps = {
  compared1: {location: '', stats: {}, districtAvg: ''},
  compared2: {location: '', stats: {}, districtAvg: ''},
  comparedAvg: {compared: '--'}
}

ComparisonWindow.propTypes = {
  compared1: PropTypes.object,
  compared2: PropTypes.object,
  comparedAvg: PropTypes.object,
}