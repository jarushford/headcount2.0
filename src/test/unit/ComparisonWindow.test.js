import React from 'react';
import ComparisonWindow from '../../ComparisonWindow';
import { shallow } from 'enzyme'

describe('Comparison Window', () => {
  it('should match the snapshot with or without data',() => {
    const compare1 = {
      location: 'Colorado',
      stats: { 2006: 0.5 }, 
      districtAvg: 0.5 
    }
    const compare2 = {
      location: 'ASPEN', 
      stats: { 2006: 0.8 }, 
      districtAvg: 0.8 
    }
    const avg = {
      'Colorado': 0.5, 
      'ASPEN': 0.8, 
      compared: 0.65
    }

    const wrapper1 = shallow(<ComparisonWindow compare1={compare1} compare2={compare2} comparedAvg={avg}/>)
    const wrapper2 = shallow(<ComparisonWindow />)

    expect(wrapper1).toMatchSnapshot()
    expect(wrapper2).toMatchSnapshot()
  })
})