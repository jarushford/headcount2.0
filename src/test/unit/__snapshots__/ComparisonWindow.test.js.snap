// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison Window should match the snapshot with or without data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonWindow
    compare1={
        Object {
            "districtAvg": 0.5,
            "location": "Colorado",
            "stats": Object {
              "2006": 0.5,
            },
          }
    }
    compare2={
        Object {
            "districtAvg": 0.8,
            "location": "ASPEN",
            "stats": Object {
              "2006": 0.8,
            },
          }
    }
    compared1={
        Object {
            "districtAvg": "",
            "location": "",
            "stats": Object {},
          }
    }
    compared2={
        Object {
            "districtAvg": "",
            "location": "",
            "stats": Object {},
          }
    }
    comparedAvg={
        Object {
            "ASPEN": 0.8,
            "Colorado": 0.5,
            "compared": 0.65,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          
           -- 
          
</h2>,
        <p>
          has
</p>,
        <h3>
          0.65
</h3>,
        <p>
          times the average attendance of
</p>,
        <h2>
          
           -- 
          
</h2>,
      ],
      "className": "comparison-window",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            " -- ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          " -- ",
          "",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "has",
        },
        "ref": null,
        "rendered": "has",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.65,
        },
        "ref": null,
        "rendered": 0.65,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "times the average attendance of",
        },
        "ref": null,
        "rendered": "times the average attendance of",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            " -- ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          " -- ",
          "",
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            
             -- 
            
</h2>,
          <p>
            has
</p>,
          <h3>
            0.65
</h3>,
          <p>
            times the average attendance of
</p>,
          <h2>
            
             -- 
            
</h2>,
        ],
        "className": "comparison-window",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              " -- ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            " -- ",
            "",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "has",
          },
          "ref": null,
          "rendered": "has",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.65,
          },
          "ref": null,
          "rendered": 0.65,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "times the average attendance of",
          },
          "ref": null,
          "rendered": "times the average attendance of",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              " -- ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            " -- ",
            "",
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comparison Window should match the snapshot with or without data 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonWindow
    compared1={
        Object {
            "districtAvg": "",
            "location": "",
            "stats": Object {},
          }
    }
    compared2={
        Object {
            "districtAvg": "",
            "location": "",
            "stats": Object {},
          }
    }
    comparedAvg={
        Object {
            "compared": "--",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          
           -- 
          
</h2>,
        <p>
          has
</p>,
        <h3>
          --
</h3>,
        <p>
          times the average attendance of
</p>,
        <h2>
          
           -- 
          
</h2>,
      ],
      "className": "comparison-window",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            " -- ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          " -- ",
          "",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "has",
        },
        "ref": null,
        "rendered": "has",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "--",
        },
        "ref": null,
        "rendered": "--",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "times the average attendance of",
        },
        "ref": null,
        "rendered": "times the average attendance of",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            " -- ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          " -- ",
          "",
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            
             -- 
            
</h2>,
          <p>
            has
</p>,
          <h3>
            --
</h3>,
          <p>
            times the average attendance of
</p>,
          <h2>
            
             -- 
            
</h2>,
        ],
        "className": "comparison-window",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              " -- ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            " -- ",
            "",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "has",
          },
          "ref": null,
          "rendered": "has",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "--",
          },
          "ref": null,
          "rendered": "--",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "times the average attendance of",
          },
          "ref": null,
          "rendered": "times the average attendance of",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              " -- ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            " -- ",
            "",
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
